jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -163,6 +162,10 @@ def _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, target_w
     else:
         sc.meta = metadata
         new_sc = sc
+
+    # Always store the input WCS to avoid getting swapped version from Spectrum1D.
+    metadata['_orig_wcs'] = wcs
```

Comment:
This actually hurts my head. I wonder if there are other cases that would break this.

In the Cubeviz mouseover logic, we hardcode the assumption of `(x, y, z)` axes order. Is this always correct, especially given astropy/specutils#999 ?